File: src/Day1.kt
fun main() {
    Timer.measure {
        val masses = Inputs.readLines("./Day1.input")
        println("Day1 Step1: " + masses.map { calc(it) }.sum())
        println("Day1 Step2: " + masses.map { recCalc(it) }.sum())
    }
}

fun calc(mass: Int): Int {
    return (mass / 3) - 2
}

fun recCalc(mass: Int): Int {
    val fuel = calc(mass)
    return when {
        fuel < 1 -> 0
        else -> fuel + recCalc(fuel)
    }
}

File: js/day1.js
var fs = require('input.js')


module.exports = {
    calc: (mass) => {
        let fuel = Math.floor (mass / 3) - 2
        return fuel <= 0 ? 0 : fuel + module.exports.calc(fuel)
    },
    
    sumCalc: (masses) => {
        console.log('m: '+ masses)
        return masses.map(module.exports.calc).reduce((a, b) => a + b, 0)
    }
}

let input = fs.readNumbers('day1.input')
console.log(module.exports.sumCalc(input))
